src/firefly/js/templates/hydra/HydraManager.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+import {take, fork} from 'redux-saga/effects';
+
+import {SHOW_DROPDOWN, SET_LAYOUT_MODE, getLayouInfo,
+        dispatchUpdateLayoutInfo, dropDownManager} from '../../core/LayoutCntlr.js';
+import {removeChartsInGroup} from '../../charts/ChartsCntlr.js';
+import {TABLE_SEARCH, TBL_RESULTS_ADDED, TABLE_REMOVE} from '../../tables/TablesCntlr.js';
+import {visRoot, dispatchDeletePlotView} from '../../visualize/ImagePlotCntlr.js';
+import {removeTablesFromGroup, getAllTableGroupIds, smartMerge} from '../../tables/TableUtil.js';
+import {getSearchInfo} from '../../core/AppDataCntlr.js';
+import ImagePlotCntlr from '../../visualize/ImagePlotCntlr.js';
+import {CHART_ADD} from '../../charts/ChartsCntlr.js';
+import {REPLACE_VIEWER_ITEMS} from '../../visualize/MultiViewCntlr.js';
+
+
+/**
+ * Configurable part of this template
+ * @typedef {object} Config
+ * @prop {Search[]} searches  list of searches.
+ * @prop {Catalog}  catalogs  additional catalogs(non-public).
+ */
+
+/**
+ * Search information.
+ * @typedef {object} Catalog
+ * @prop {string} url  url of the additional catalog
+ */
+
+/**
+ * This event manager is custom made for HydraViewer.
+ */
+export function* hydraManager() {
+
+    yield fork(dropDownManager);        // start the dropdown manager
+    while (true) {
+        const action = yield take([
+            SHOW_DROPDOWN, SET_LAYOUT_MODE, CHART_ADD,
+            TABLE_SEARCH, TBL_RESULTS_ADDED, TABLE_REMOVE,
+            ImagePlotCntlr.PLOT_IMAGE, ImagePlotCntlr.PLOT_IMAGE_START, ImagePlotCntlr.DELETE_PLOT_VIEW, REPLACE_VIEWER_ITEMS
+        ]);
+
+        /**
+         * This is the current state of the layout store.  Action handlers should return newLayoutInfo if state changes
+         * If state has changed, it will be dispatched into the flux.
+         * @type {LayoutInfo}
+         * @prop {boolean}  layoutInfo.showForm    show form panel
+         * @prop {boolean}  layoutInfo.showTables  show tables panel
+         * @prop {boolean}  layoutInfo.showXyPlots show charts panel
+         * @prop {boolean}  layoutInfo.showImages  show images panel
+         * @prop {Object}   layoutInfo.images      images specific states
+         * @prop {string}   layoutInfo.images.activeTableId  last active table id that images responded to
+         * @prop {string}   layoutInfo.currentSearch  the current selected search
+         *
+         */
+        var layoutInfo = getLayouInfo();
```

Comment:
I have already defined a LayoutInfo typedef in core-typedefs.jsdoc.  I will move all of the common prop there as well and only leave behind the ones that is only applicable for hydraManager here.